### DISCUSSION TITLE
Disable running status in HairlessMIDI_Interface?

### DISCUSSION THREAD

**Discussion by zigmhount** [tttapa/Control-Surface#1102] [Q&A] [answered]
Hi everyone,

I'm dabbling with Arduino and built a MIDI pedal a few years ago with an Arduino Nano, so I had to use Control-Surface's HairlessMIDI_Interface instead of USBMIDI and I'm listening to it with [ttymidi](http://www.varal.org/ttymidi/). Unfortunately ttymidi doesn't support running status, and I'm suspecting that it might cause an issue in my setup (some `dropping unexpected MIDI byte` errors from Pure Data).

I found [here](https://github.com/projectgus/hairless-midiserial/issues/16) a suggestion to prevent Hairless to use running status, but I couldn't find anything about this in Control-Surface's documentation. Is it even relevant for Control-Surface?

Thanks in advance for your help and guidance :)

**Comment by tttapa** [comment, marked as answer]
Control Surface doesn't use running status on transmit (it does support receiving running status messages).  
Which MIDI bytes does Pure Data mark as "unexpected"? Do you have some kind of log of the MIDI messages?

**Comment by zigmhount** [comment]
Thanks for the quick reply! Good to know that running status is not the issue there then, thanks a lot.

I've also been struggling to find anything online about reasons for Pure Data to drop MIDI bytes. Unfortunately I lost the logs from yesterday, and I can't seem to reproduce it today! My pedal has latching buttons and just sends `CC 1 127` followed by `CC 1 0` on channel 1, which were correctly read by `aseqdump`, but Pure Data was dropping a dozen of bytes, which I'm guessing may have been the full CC message.
I'll share it when I get the issue again. That said, the fact that it is not happening today is probably pointing at a performance-related issue. Do you know whether a small value for Jack's buffer size or Pipewire's `clock.quantum` can cause Pd to drop bytes?
